commit 299904b493ebc17b586ae4df3cae8312331ae8dc
Author:     Dan Davison <dandavison7@gmail.com>
AuthorDate: Sun Jun 30 17:01:26 2019 -0400
Commit:     Dan Davison <dandavison7@gmail.com>
CommitDate: Sun Jun 30 17:06:07 2019 -0400

    Classify themes as light vs dark in --list-themes output

diff --git a/src/assets.rs b/src/assets.rs
index 2f327c6..b5adafb 100644
--- a/src/assets.rs
+++ b/src/assets.rs
@@ -116,16 +116,3 @@ pub fn list_languages() -> std::io::Result<()> {
 
     Ok(())
 }
-
-pub fn list_themes() -> std::io::Result<()> {
-    let assets = HighlightingAssets::new();
-    let themes = &assets.theme_set.themes;
-    let stdout = io::stdout();
-    let mut stdout = stdout.lock();
-
-    for (theme, _) in themes.iter() {
-        writeln!(stdout, "{}", theme)?;
-    }
-
-    Ok(())
-}
diff --git a/src/main.rs b/src/main.rs
index e2e46f4..413876a 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -5,7 +5,7 @@ mod parse_diff;
 use std::io::{self, BufRead, ErrorKind, Read, Write};
 use std::process;
 
-use assets::{HighlightingAssets, list_languages, list_themes};
+use assets::{HighlightingAssets, list_languages};
 use console::strip_ansi_codes;
 use structopt::StructOpt;
 use syntect::parsing::SyntaxReference;
@@ -218,3 +218,25 @@ fn compare_themes(assets: &HighlightingAssets) -> std::io::Result<()> {
 
     Ok(())
 }
+
+
+pub fn list_themes() -> std::io::Result<()> {
+    let assets = HighlightingAssets::new();
+    let themes = &assets.theme_set.themes;
+    let stdout = io::stdout();
+    let mut stdout = stdout.lock();
+
+    writeln!(stdout, "Light themes:")?;
+    for (theme, _) in themes.iter() {
+        if paint::is_light_theme(theme) {
+            writeln!(stdout, "    {}", theme)?;
+        }
+    }
+    writeln!(stdout, "Dark themes:")?;
+    for (theme, _) in themes.iter() {
+        if !paint::is_light_theme(theme) {
+            writeln!(stdout, "    {}", theme)?;
+        }
+    }
+    Ok(())
+}
diff --git a/src/paint.rs b/src/paint.rs
index c7d1776..5e05aa6 100644
--- a/src/paint.rs
+++ b/src/paint.rs
@@ -12,6 +12,10 @@ pub const LIGHT_THEMES: [&str; 4] = [
     "ansi-light",
 ];
 
+pub fn is_light_theme(theme: &str) -> bool {
+    LIGHT_THEMES.contains(&theme)
+}
+
 const LIGHT_THEME_PLUS_COLOR: Color = Color {
     r: 0xd0,
     g: 0xff,
